File: lab5/cpp/main.cpp
#include "Classes.h"

int main() {
	srand(time(NULL));
	
	TMatrix* arr[]{new SquareMatrix2Order(), new SquareMatrix3Order()};
	double sum = 0;

	for (auto elem : arr) {
		cout << "\nMatrix:\n";
		elem->printMatrix();
		cout << "\nMatrix determinant: " << elem->getDeterminant() << endl;
		cout << "Sum of elements: " << elem->getSumOfElements() << endl;
		sum += elem->getDeterminant();
		if (dynamic_cast<SquareMatrix3Order*>(elem)) {
			sum += elem->getSumOfElements();
		}
	}

	cout << "\n\nS = " << sum << endl;

	system("pause");
	return 0;
}

File: lab5/python/main.py
import classes

listMatrix = [classes.SquareMatrix2Order(), classes.SquareMatrix3Order()]
s = 0

for elem in listMatrix:
    print("\nMatrix:")
    elem.printMatrix()
    print(f"\nMatrix determinant: {elem.getDeterminant()}")
    s += elem.getDeterminant()
    print(f"Sum of elements Matrix: {elem.getSumOfElements()}")
    if isinstance(elem, classes.SquareMatrix3Order):
        s += elem.getSumOfElements()

print(f"\nS = {s}")
